code/client/launcher/Console.Logging.cpp

```diff
@@ -172,57 +177,85 @@ struct LoggerInit
 
 static LoggerInit logger;
 
-extern "C" DLL_EXPORT void AsyncTrace(const char* string)
+extern "C" DLL_EXPORT void AsyncTrace(const char* str)
 {
 	std::string threadName = GetThreadName();
 
-	g_logPrintQueue.push({ threadName, string });
+	g_logPrintQueue.push({ threadName, str });
 	g_logCondVar.notify_all();
 }
 
 void InitLogging()
 {
 	SetThreadName(-1, "MainThrd");
 
-	if (OpenMutex(SYNCHRONIZE, FALSE, L"CitizenFX_LogMutex_Mod") == nullptr)
+	const auto handle_deleter = [](HANDLE handle){ if (handle != INVALID_HANDLE_VALUE) CloseHandle(handle); };
+	using UniqueHandle = std::unique_ptr<std::remove_pointer<HANDLE>::type, decltype(handle_deleter)>;
```

Comment:
Is using `decltype` to get the type of a captureless lambda considered fine? Perhaps there's more cases like this straddled throughout code that could benefit from pulling in [WIL](https://github.com/microsoft/wil).

Reply:
Yes and what for WIL - totally Yes, if it is ok to add another dependency to the project.